ykrt/src/compile/jitc_yk/trace_builder.rs

```diff
@@ -688,10 +716,25 @@ impl TraceBuilder {
             }
             Ok(())
         } else {
-            // We've returned out of the function that started tracing, which isn't allowed.
-            Err(CompilationError::General(
-                "returned from function that started tracing".into(),
-            ))
+            if let TraceKind::Sidetrace(_) = self.jit_mod.tracekind() {
+                // Even though we currently catch side-traces that leave the main interpreter loop
+                // in `mt.rs`, it can happen that a trace re-enters the main interpreter loop after
+                // leaving it briefly without encountering a non-null location. As we only check
+                // the frame addresses at non-null locations, this special case slips through and
+                // we have to handle it here.
+                // FIXME: Abort side-traces the same way we do normal traces by emitting a return
```

Comment:
What does this FIXME mean? Why would we abort side-traces with a `return`?

Reply:
Sorry, that shouldn't say "abort". I meant that we should do the same that we now do for normal traces that return from the interpreter: emit a return. Will fix the comment.

Reply:
[7b198c7](https://github.com/ykjit/yk/pull/1808/commits/7b198c741db495bdec93a4833eca49cb6e935a76)